internal/cmd/volume/snapshot/list/list.go

```diff
@@ -0,0 +1,198 @@
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`List all snapshots`,
+				"$ stackit volume snapshot list"),
+			examples.NewExample(
+				`List snapshots with a limit of 10`,
+				"$ stackit volume snapshot list --limit 10"),
+			examples.NewExample(
+				`List snapshots filtered by label`,
+				"$ stackit volume snapshot list --label-selector key1=value1"),
+		),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := context.Background()
+			model, err := parseInput(params.Printer, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(params.Printer, params.CliVersion)
+			if err != nil {
+				return err
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("list snapshots: %w", err)
+			}
+
+			// Check if response is empty
+			if resp.Items == nil || len(*resp.Items) == 0 {
+				projectLabel, err := projectname.GetProjectName(ctx, params.Printer, params.CliVersion, cmd)
+				if err != nil {
+					params.Printer.Debug(print.ErrorLevel, "get project name: %v", err)
+					projectLabel = model.ProjectId
+				}
+				params.Printer.Info("No snapshots found for project %q\n", projectLabel)
+				return nil
+			}
+
+			snapshots := *resp.Items
+
+			// Apply limit if specified
+			if model.Limit != nil && int(*model.Limit) < len(snapshots) {
+				snapshots = snapshots[:*model.Limit]
+			}
+
+			return outputResult(params.Printer, model.OutputFormat, snapshots)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Int64(limitFlag, 0, "Maximum number of entries to list")
+	cmd.Flags().String(labelSelectorFlag, "", "Filter snapshots by labels")
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	limit := flags.FlagToInt64Pointer(p, cmd, limitFlag)
+	if limit != nil && *limit < 1 {
+		return nil, &errors.FlagValidationError{
+			Flag:    limitFlag,
+			Details: "must be greater than 0",
+		}
+	}
+
+	labelSelector := flags.FlagToStringPointer(p, cmd, labelSelectorFlag)
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		Limit:           limit,
+		LabelSelector:   labelSelector,
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiListSnapshotsRequest {
+	req := apiClient.ListSnapshots(ctx, model.ProjectId)
+	if model.LabelSelector != nil {
+		req.LabelSelector(*model.LabelSelector)
```

Comment:
The request needs to be set with the return of the `LabelSelector()`. Otherwise it wouldn't be used
```suggestion
		req = req.LabelSelector(*model.LabelSelector)
```

Reply:
To check if this work, a test can be added to `TestBuildRequest()`